### PR TITLE
Convert lib/utils to slog

### PR DESCRIPTION
**Pull request by rosstimothy**
This only converts the places in the utils package that were logging with logrus, but it does not fully remove the logrus package as a dependency. The package is home to the logrus text formatter which needs to exist until slog is the only logging mechanism in use.

### CODE REVIEW

File: lib/utils/addr.go [hunk 1]
```diff
@@ -290,7 +291,7 @@ func GuessHostIP() (ip net.IP, err error) {
 	for _, iface := range ifaces {
 		ifadrs, err := iface.Addrs()
 		if err != nil {
-			log.Warn(err)
+			slog.WarnContext(context.Background(), "Unable to get addresses for interface", "interface", iface.Name, "error", err)
```
Comment (kopiczko) [inline lib/utils/addr.go:294]: Shouldn't this be context.TODO?

File: lib/utils/loadbalancer.go [hunk 2]
```diff
@@ -156,7 +155,7 @@ func (l *LoadBalancer) AddBackend(b NetAddr) {
 	l.Lock()
 	defer l.Unlock()
 	l.backends = append(l.backends, b)
-	l.Debugf("Backends %v.", l.backends)
+	l.logger.DebugContext(l.ctx, "Backends updated", "backends", l.backends)
```
Comment (kopiczko) [inline lib/utils/loadbalancer.go:158]: "Backends added"?

File: lib/utils/loadbalancer.go [hunk 3]
```diff
@@ -242,7 +243,7 @@ func (l *LoadBalancer) Serve() error {
 func (l *LoadBalancer) forwardConnection(conn net.Conn) {
 	err := l.forward(conn)
 	if err != nil {
-		l.Warningf("Failed to forward connection: %v.", err)
+		l.logger.WarnContext(l.ctx, "Failed to forward connection", "error", err)
```
Comment (kopiczko) [inline lib/utils/loadbalancer.go:246]: Won't this print stack trace now? Is it desired?

File: lib/utils/registry/registry_windows.go [hunk 4]
```diff
@@ -34,14 +35,17 @@ func GetOrCreateRegistryKey(name string) (registry.Key, error) {
...
-			log.Debugf("Can't create registry key %v: %v", name, err)
+			slog.DebugContext(context.Background(), "Can't create registry key",
+				"key_name", name,
+				"error", err,
+			)
 			return reg, err
 		}
 	case err != nil:
-		log.Errorf("registry.OpenKey returned error: %v", err)
+		slog.ErrorContext(context.Background(), "registry.OpenKey returned error", "error", err)
```
Comment (kopiczko) [inline lib/utils/registry/registry_windows.go:48]: Same question about stack trace

**Review by kopiczko — COMMENTED** [review]
Mostly LGTM, but those context.Backgroud() calls outside test files are a bit concerning. It may be less obvious to replace them with a proper context even if the calling function eventually takes it as an argument. But possibly that's something already discussed?

**Review by kopiczko — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Pull request merged**